spec/CodeCoverageQualityGate/CodeCoverageQualityService.spec.ts

```diff
@@ -0,0 +1,495 @@
+        expect(typeNew.type).toEqual(TypeEnum.MODIFIED);
+
+        settings.type = 'Overall';
+        typeTotal = createQualityGate(settings, mockWebApi);
+        expect(typeTotal.type).toEqual(TypeEnum.OVERALL);
+        settings.type = 'Modified';
+        typeNew = createQualityGate(settings, mockWebApi);
+        expect(typeNew.type).toEqual(TypeEnum.MODIFIED);
+    });
+
+    it('Set quality gate thresholds', () => {
+        settings.threshold="10";
+        let codeCoverageQualityService = createQualityGate(settings, mockWebApi);
+        expect(codeCoverageQualityService.threshold).toEqual(10);
+
+        settings.threshold="10.11";
+        codeCoverageQualityService = createQualityGate(settings, mockWebApi);
+        expect(codeCoverageQualityService.threshold).toEqual(10.11);
+
+        settings.threshold="101";
+        codeCoverageQualityService = createQualityGate(settings, mockWebApi);
+        expect(codeCoverageQualityService.threshold).toEqual(100.0);
+        expect(tl.warning).toHaveBeenCalledWith("The threshold value \'101\' is more than 100, the value is set to 100.0");
+
+        settings.threshold="-1";
+        codeCoverageQualityService = createQualityGate(settings, mockWebApi);
+        expect(codeCoverageQualityService.threshold).toEqual(0.0);
+        expect(tl.warning).toHaveBeenCalledWith("The threshold value \'-1\' is less than 0, the value is set to 0.0");
+
+        settings.threshold = "a10";
+        codeCoverageQualityService = createQualityGate(settings, mockWebApi);
+        expect(codeCoverageQualityService.threshold).toEqual(0.0);
+        expect(tl.warning).toHaveBeenCalledWith("Invalid threshold value \'a10\', using default value 0.0");
+    });
+
+    it('Set quality gate status', () => {
+        settings.buildStatus = 'unknown';
+        let unknown = createQualityGate(settings, mockWebApi);
+        expect(unknown.buildStatus).toEqual(BuildStatusEnum.FAILED);
+        expect(tl.warning).toHaveBeenCalledWith('Invalid value for \'buildStatus\': unknown, using default value \'Failed\'');
+
+        settings.buildStatus = 'unstable';
+        let unstable = createQualityGate(settings, mockWebApi);
+        expect(unstable.buildStatus).toEqual(BuildStatusEnum.UNSTABLE);
+        settings.buildStatus = 'failed';
+        let failed = createQualityGate(settings, mockWebApi);
+        expect(failed.buildStatus).toEqual(BuildStatusEnum.FAILED);
+
+        settings.buildStatus = 'Unstable';
+        unstable = createQualityGate(settings, mockWebApi);
+        expect(unstable.buildStatus).toEqual(BuildStatusEnum.UNSTABLE);
+        settings.buildStatus = 'Failed';
+        failed = createQualityGate(settings, mockWebApi);
+        expect(failed.buildStatus).toEqual(BuildStatusEnum.FAILED);
+    });
+
+    describe('When evaluate quality gate and', () => {
+        let setUpQualityGate = (currentFileEntry: any, currentBuildArtifact?: any, referenceBuildArtifact?: any, referenceFileEntry?: any) => {
```

Comment:
The order of params may cause some confuse, for better readability, use order like: (currentFileEntry, referenceFileEntry, currentBuildArtifact,  referenceBuildArtifact) or (currentFileEntry, currentBuildArtifact, referenceFileEntry,  referenceBuildArtifact). When some optional param is not needed, pass undefined.